articles/setting-up-cicd-for-python-packages-using-github-actions/index.md

```diff
@@ -0,0 +1,279 @@
+          - name: Install dependencies
+            run: |
+              python -m pip install --upgrade pip
+              pip install flake8
+          - name: Lint with flake8
+            run: |
+              # stop the build if there are Python syntax errors or undefined names
+              flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+              # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+              flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+          - name: Build binary wheel and a source tarball
+            run: python setup.py sdist
+          - name: Publish distribution to Test PyPI
+            uses: pypa/gh-action-pypi-publish@master
+            with:
+              password: ${{ secrets.test_pypi_password }}
+              repository_url: https://test.pypi.org/legacy/
+    ```
+
+Few things to note about the above workflow:
+
+1. We have just a single `build-n-publish` job which runs on `ubuntu-18.04`. 
+2. Then, we checkout the project into the ubuntu environment and setup our python distribution (Python 3.7)
+3. Then, we install dependencies needed for the package and test it against a `flake8` linter.
+4. Once done, we will produce a source distribution from the packages coded. We do this using the `python [setup.py](http://setup.py) sdist` command.
+5. The last step uses `pypa/gh-action-pypi-publish` GitHub Action to upload contents of the dist/ folder into TestPyPI unconditionally.  It also used the secrets declared and defined in the previous section. 
+
+Below is the final directory structure:
+
+```yaml
+.
+├── .github
+│   └── workflows
+│       └── python-package.yml
+├── .gitignore
+├── LICENSE
+├── README.md
+├── setup.py
+└── src
+    ├── __init__.py
+    ├── logic.py
+    └── main.py
+```
+
+Once this is achieved, push the code to the repository. Then navigate to the **Actions** tab and see something similar to the screenshot below:
+
+![actions.png](/engineering-education/setting-up-cicd-for-python-packages-using-github-actions/actions.png)
+
+### Things To Note
+
+- If you're facing *"the user <username> is not allowed...",* change the name of the package in `[setup.py](http://setup.py)` to `<username>_timechecker`.
+- If you face an indentation error, in the pipeline, follow the error on the line flagged and try to fix the indentation error. This is part of CI/CD.
+
+## Testing Python Package Locally
+
+To test the package locally, execute the following command locally:
+
+```bash
+pip install -i https://test.pypi.org/simple/ timechecker
+timechecker --location Algiers
+timechecker --zone EST
+```
+
+## Conclusion
+
+At the end of this tutorial, we have built a python application, packaged it in GitHub Action, and deployed it to Test PyPI using GitHub Actions. I hope this guide helps anyone in the future when building their Python packages and wanting to push them using GitHub Actions.
```

Comment:
Add another section called `Further reading`, and reference the sources that you have used, and also suggest what other articles related to this topic, can be read.

Reply:
The main article I would've loved to add here hasn't been published yet. 
#1217

Reply:
I've added this section though

Reply:
No worries. You can update the article after the other article gets published.